### ISSUE TITLE
Update to python3

### ISSUE THREAD

**Issue by tophee** [mschmidtkorth/alfred-clickup-msk#35]
**What is the purpose of your feature request?**
With macOS 12.3, python 2 has been removed from macOS so that this workflow no longer works

**Describe the solution you'd like**
Update the workflow to python3

**Comment by mschmidtkorth** [comment]
I'd have done it already @tophee, unfortunately that would require a larger rework as the [core library](https://github.com/deanishe/alfred-workflow) depends on Python 2.7.
You can always install Python 2.7 to your OS.

**Comment by tophee** [comment]
Do you mean you have tried it and it was too much work?

I would like to avoid installing a very outdated python version on my system (also because google suggests all kinds of issues when doing so), but I completely understand and respect that you don't want to invest time in updating the workflow. But have you considered switching the core library to https://github.com/NorthIsUp/alfred-workflow-py3 ? It seems to work rather well according to [some](https://github.com/deanishe/alfred-workflow/issues/97#issuecomment-1073251500), although [it depends](https://github.com/deanishe/alfred-workflow/issues/97#issuecomment-1073254615) on what exactly you use.

**Comment by mschmidtkorth** [comment]
No, I'm saying that the ClickUp workflow uses a 3rd-party library that I am not developing. This library requires Python 2 and is used to interact with Alfred in an easier way. 

The ClickUp workflow itself could easily be migrated to Python 3 and I'd happily do so, though without the library being migrated as well this would be of limited use.

I was not aware that there is a fork that works with Python 3, thanks for mentioning - that's something I'll definitely be looking into, though not in the next fews weeks as my agenda is quite busy. It would also require extensive testing.

Of course, I'd gladly accept any PR doing the same.

With that information, I'll reopen this issue.

**Comment by Peeekay** [comment]
@mschmidtkorth Did you ever find fix or replacement?

**Comment by ihscmagill** [comment]
Has anyone gotten this workflow working by installing Python 2? I followed the instructions here, but it seems there is more work needed.

https://www.alfredapp.com/help/kb/python-2-monterey/

**Comment by Peeekay** [comment]
> Has anyone gotten this workflow working by installing Python 2? I followed the instructions here, but it seems there is more work needed.
> 
> https://www.alfredapp.com/help/kb/python-2-monterey/

Never tried installing Python 2. Sorry.

**Comment by mschmidtkorth** [comment]
@ihscmagill Python 2 should be straightforward without special setup needed. What does not work for you?

@Peeekay I haven't made any progress. I'm currently mostly using Raycast (and monitoring their development) instead of Alfred and therefore putting my projects on hold for the time being until I've made a decision.

**Comment by ihscmagill** [comment]
@mschmidtkorth I receive the following error

ERROR: ClickUp[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.0FD8C38D-8379-4A84-AF8B-664B81882CB0%3EBA60D8FC-3366-47F1-BDF9-824C1E5C1E39)] Code 127: /Users/christianmagill/Library/Caches/com.runningwithcrayons.Alfred/Workflow Scripts/4568F254-60AD-4B95-9021-3178443125FB: line 1: /usr/bin/python: No such file or directory

**Comment by ihscmagill** [comment]
@mschmidtkorth A bit heavy-handed but I think I got it to work by replacing all instances of `/usr/bin/python` with `/opt/homebrew/bin/python`

**Comment by RvDelft** [comment]
Note: If there is a new version. Al this work will be undone. So keep that in mind!

**Comment by cdouglasnet** [comment]
I know that this has been dormant for a long time...

There is now a fork of this project that is up and running (Python 3) and has fixed many of the issues here for those wanting to get ClickUp working with Alfred again. I tried working on it based on the work of @WreckItR but didn't get very far. Greg Flint with Four13 Digital is the one who got it working.
Here is his fork:

https://github.com/four13co/alfred-clickup-four13

Thanks @mschmidtkorth for all that you have done to get this project going and started.
Thanks @gregflint for working on this and getting things running on Newer Macs.
Thanks @WreckItR for working on this as well.

Hopefully this is helpful.

**Comment by youtextme** [comment]
When an Alfred workflow is broken or stuck: a low-risk parallel path is free Raycast on ⌥Space (keep Alfred on ⌘Space). Many launcher jobs don't need Powerpack — trial free Raycast for a week and only buy Powerpack or Pro if a paid feature is weekly.

https://raycast.com/?via=girish
Affiliate — I earn a commission if you upgrade to Pro.
